### ISSUE TITLE
[enhancement] Autosubmit stop $expid ?

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1190]
In GitLab by @ainagaya on Dec 29, 2023, 15:51

Hello @dbeltrankyl and @kinow,

I thought that maybe it would be a good idea to have an "autosubmit kill $expid" option to kill the experiments that are executed with nohup. That option would first cancel the ongoing queueing/running jobs in the HPCs or wherever and then kill autosubmit itself (this sounds a bit weird). Sometimes is a bit tedious when you want to stop an experiment for some reason and you need to cancel the jobs one by one. Or sometimes users forget that killing autosubmit does not kill the remote jobs and wastes some resources. Maybe you already discussed about this but I couldn't find it!

What do you think? :santa: :christmas_tree: 

()

**Comment by kinow** [comment]
+1 from me, @ainagaya ! That's available in cwltool, cylc, ecflow, ...

I can expand on how that works in Cylc, and maybe that could give us some more ideas to refine how it should work.

- `cylc stop $workflow_id`

This stops a workflow. It will wait for all active tasks to stop and then exists.

But if you want to kill all active tasks, then you call `cylc stop $workflow_id --kill`. That will kill active tasks and exit.

Or, if you want to exit now without waiting for the tasks: `cylc stop $workflow_id --now` (there's also `--now --now` repeated, to really exit immediately without waiting for some internal work to complete).

- `cylc kill $task_id`

This kills a job/task in Cylc.

I wonder if we should call `autosubmit stop $workflow` instead of `kill`, @ainagaya , just to avoid confusion for anyone familiar with Cylc. In ecFlow `ecflow_client kill` is also used to “Kills the job associated with the node.”.

I think we also do not have a command for that in Autosubmit. Say you identified something is wrong with a chunk or member execution (corrupted file?). If the job is still running in Cylc, you kill that job, pause the workflow (or wait for it to stall), fix the file, and rerun the job (similar to our setstatus in Autosubmit).

:santa: :christmas_tree:

**Comment by kinow** [comment]
In GitLab by @ainagaya on Jan 11, 2024, 09:28

mentioned in issue digital-twins/de_340/workflow#113

**Comment by kinow** [comment]
And `cylc stop --kill $workflow_id` stops the workflow and kills its jobs. Which avoids issues like a user forgetting to kill the running jobs after killing the workflow, and then re-submitting and ending up with multiple slurm jobs running by accident, consuming unnecessary resources and potentially preventing the more recent run from proceeding (which happened recently in an issue @ainagaya is in copy -- i.e. we have a practical use case to solve in another project with this enhancement).

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jan 12, 2024, 10:48

Hello all , 

Thanks for the request. 

Yes, I think it will be a nice addition. I prefer the name that Bruno proposed (autosubmit stop).

I would add: 


``` bash
autosubmit stop $expid # same behaviour as control + c
autosubmit stop $expid --kill # stops + kills active jobs and set them to failure
autosubmit stop $expid --kill (status) # stops + kills active jobs and set them to designed status
```

What do you think?

**Comment by kinow** [comment]
In GitLab by @ainagaya on Mar 5, 2024, 17:03

mentioned in issue digital-twins/de_340/workflow#248

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 13, 2024, 16:23

This command is now avaliable in the hubs (4.1.6) 

@ainagaya , syntax in : https://autosubmit.readthedocs.io/en/master/userguide/run/index.html#how-to-stop-the-experiment

**Comment by kinow** [comment]
In GitLab by @ainagaya on May 13, 2024, 16:27

That's great! Can't wait to test it! Thanks Dani :)

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Jun 11, 2024, 11:31

Implemented in 4.1.6, if there are issues with it's usage ( in 4.1.7, if possible ) open a new issue
